lib/Analysis/Allocation.cpp

```diff
@@ -112,6 +112,20 @@ SmallVector<unsigned> getScratchConfigForReduce(triton::ReduceOp op) {
   return smemShape;
 }
 
+// only valid for scalar
```

Comment:
Would be better phrased as "// TODO: extend beyond scalars". This will make it easier to track this when we scan through TODOs in the code

Reply:
The comments is not a TODO. Maybe i should make me more clear. The comments is to tell we only use shared memory in scalar cases.

Reply:
But shouldn't we also use it for e.g. slices?

Reply:
I see and I will add one TODO.

Reply:
Add one TODO. Done.